docs/design/uptime-reputation-calculation.md

```diff
@@ -0,0 +1,177 @@
+
+The problem with the current uptime _SN_ reputation score calculation is that it isn't based on the actual time that a node is being offline, by cause of to the nature of the chosen model, causing that an _SN_ will get disqualified due its number of uptime checks success and failure which implies that new nodes can get disqualified having less uptime check failures than veteran nodes.
+
+While we understand that an _SN_ that has remained in the network with a low uptime check failures could be considered a more reliable _SN_ than new ones which recently joined, we see, at the same time, that new _SN_ shouldn't be harsher penalized than veteran nodes due to uptime check failures because:
+
+1. Any _SN_ (veteran and new ones) requires some time to be offline for maintenance and both should have the same maximum amount of time for it.
+1. The end goal of the uptime checks is to ensure that the network has certain availability and retrievability and uptime check failures affect to them no matter on how long the _SN_ has remained in the network and how much data an _SN_ has, hence node with more or less data must be treated equally.
+
+This design doc is focused on how to design and implement a fair, simple and understandable uptime reputation calculation after our initial approach didn't behave as expected, causing the disqualification of several nodes without having a proof that they have failed enough uptime checks that the _SNs_ were offline long time enough which put in danger the minimum level of the network availability and retrievability.
+
+## Design
+
+The new process for doing the uptime checks is based on once an _SN_ is selected for having the check, if the _SN_ fails the initial check, then it's tracked to periodically receive uptime checks until (a) it successfully passes an uptime check or (b) gets disqualified due to be offline the maximum allowed by the established number of days (henceforh uptime check period) or it isn't needed to be checked anymore because it has been excluded from the network by any reason.
+
+Once an _SN_ has failed the first uptime check, the _SN_ is considered offline and the subtotal time, which the _SN_ is considered offline, is the time elapsed between the first and the last failed uptime check. The total offline time of an _SN_ is the accumulation of the subtotals that the _SN_ has had during the ongoing uptime check period and when such total surpasses the maximum permitted percentage offline time, the _SN_ will get disqualified.
+
+The maximum percentage of time that an _SN_ can be offline will be established with the network commitment in terms of availability and retrievability during one month.
+
+The process is divided in 2 parts:
+
+1. Selecting _SNs_ for receiving uptime checks.
```

Comment:
Maybe we can add the uptime auditor here: satellite tries to reach nodes we haven't heard from in an hour and considers uptime audit failures for reputation. If this is out of scope, maybe we should create another design doc

Reply:
I would address the audit uptime failures in another doc, considering that this one will be renamed to be only uptime tracking/calculation.
Nonetheless, I will take a look at the audit code.